File: Practice/src/introduction_coding_tests/SafeZone.js
function solution(board) {
    var answer = 0;
    const arr = [];
    for (let i = 0; i < board.length + 2; i++) {
        arr.push(Array(board.length + 2).fill(false));
    }
    for (let i = 0; i < board.length; i++) {
        for (let j = 0; j < board[0].length; j++) {
            if (board[i][j] == 1) {
                for (let k = 0; k < 3; k++) {
                    for (let l = 0; l < 3; l++) {
                        arr[k + i][l + j] = true;
                    }
                }
            }
        }
    }
    for (let i = 1; i <= board.length; i++) {
        for (let j = 1; j <= board.length; j++) {
            if (!arr[i][j]) answer++;
        }
    }
    return answer;
}

File: Practice/src/introduction_coding_tests/SafeZone.java
package introduction_coding_tests;

public class SafeZone {
//	안전지대
	public int solution(int[][] board) {
		int[][] arr = new int[board.length + 2][board[0].length + 2];
		int answer = 0;
		for (int i = 0; i < board.length; i++) {
			for (int j = 0; j < board[i].length; j++) {
				if (board[i][j] == 1) {
					arr[i][j] = 1;
					arr[i + 1][j] = 1;
					arr[i + 2][j] = 1;
					arr[i][j + 1] = 1;
					arr[i + 1][j + 1] = 1;
					arr[i + 2][j + 1] = 1;
					arr[i][j + 2] = 1;
					arr[i + 1][j + 2] = 1;
					arr[i + 2][j + 2] = 1;
				}

			}
		}
		for (int i = 1; i <= board.length; i++) {
			for (int j = 1; j <= board[i].length; j++) {
				if (arr[i][j] == 0)
					answer++;
			}
		}
		return answer;
	}
}